packages/nc-gui/composables/useUIPermission/index.ts

```diff
@@ -11,6 +11,14 @@ const hasPermission = (role: Role | ProjectRole, hasRole: boolean, permission: P
 
   if (isString(rolePermission) && rolePermission === '*') return true
 
+  // todo: type correction
+  if ('include' in rolePermission && rolePermission.include) {
+    return rolePermission.include[permission]
+  }
+  if ('exclude' in rolePermission && rolePermission.exclude) {
+    return !rolePermission.exclude[permission]
+  }
```

Comment:
I'm not sure if this logic really works - 

Previously, if we checked a permission and the user had the role `Creator`, it would give access to any permission as it had the `*` wildcard permission assigned.
Now it has *no* permissions assigned, except for permissions that are *excluded*.

So shouldn't a creator now have no permissions at all?
Same for the `Owner` role.

Reply:
The `excluded` means everything excluding those specified permissions, it's like a wildcard with some limitation.

Reply:
I see.

Could you add a comment explaining the logic behind that so it's obvious to idiots like me? 😄